src/Database/LSMTree/Internal.hs

```diff
@@ -1161,7 +1164,12 @@ open sesh label override snap = do
                 snapPath
                 FS.ReadMode $ \h ->
                   FS.hGetAll (sessionHasFS seshEnv) h
-        let (label', snappedLevels, conf) = read $ BSC.unpack $ BSC.toStrict $ bs
+
+        snapMetaData <- case decodeSnapshotMetaData bs of
+          Left e  -> throwIO (ErrSnapshotDeserialiseFailure e snap)
+          Right x -> pure x
+        let SnapshotMetaData label' conf snappedLevels = snapMetaData
+
```

Comment:
Should we take any precautions against corrupt snapshots, e.g. violating the invariants we have on number of runs?

Reply:
Eventually, the snapshot metadata file will get a checksum, so that would protect against accidental corruption -- we don't have to take precautions against intentional corruption. Does that sound sufficient?

Reply:
Mostly. It would be an additional safety net, but probably not needed.